e2e/ui/Enterprise SSO/setup_link.spec.ts

```diff
@@ -0,0 +1,154 @@
+import { Locator, Page, expect, test as baseTest } from '@playwright/test';
+import { Portal } from 'e2e/support/fixtures';
+
+const TEST_SETUPLINK_REDIRECT_URL = 'http://localhost:3366';
+const TEST_SETUPLINK_DEFAULT_REDIRECT_URL = 'http://localhost:3366/login/saml';
+const TEST_SETUPLINK_MOCKSAML_ORIGIN = process.env.MOCKSAML_ORIGIN || 'https://mocksaml.com';
+const TEST_SETUPLINK_MOCK_METADATA_URL = `${TEST_SETUPLINK_MOCKSAML_ORIGIN}/api/saml/metadata`;
+const TEST_SETUPLINK_URL_LABEL_SELECTOR =
+  'Share this link with your customers to allow them to set up the integrationClose';
+const TEST_SETUPLINK_ADMIN_URL = '/admin/sso-connection/setup-link';
+const TEST_SETUPLINK_ADMIN_CONNECTION = '/admin/sso-connection';
+
+type MyFixtures = {
+  portal: Portal;
+};
+
+export const test = baseTest.extend<MyFixtures>({
+  portal: async ({ page }, use) => {
+    const portal = new Portal(page);
+    await portal.doCredentialsLogin();
+    await use(portal);
+  },
+});
+
+async function retryForCondition(condition: Promise<any>, retries = 10, retryInterval = 500) {
+  while (retries) {
+    if (await condition) {
+      break;
+    } else {
+      setTimeout(() => {
+        retries -= 1;
+      }, retryInterval);
+    }
+  }
+}
+
+async function getRowByTenantProduct(
+  page: Page,
+  tenant: string,
+  product: string,
+  tenantProductIndexes: Array<number>
+): Promise<Locator | string> {
+  const rows = await page.getByRole('row').all();
```

Comment:
Instead of using this logic here, can we simply assert either the name of the connection (unique) or the tenant/product in the page. All locator can result in test being flaky.
> [locator.all()](https://playwright.dev/docs/api/class-locator#locator-all) does not wait for elements to match the locator, and instead immediately returns whatever is present in the page. When the list of elements changes dynamically, [locator.all()](https://playwright.dev/docs/api/class-locator#locator-all) will produce unpredictable and flaky results. When the list of elements is stable, but loaded dynamically, wait for the full list to finish loading before calling [locator.all()](https://playwright.dev/docs/api/class-locator#locator-all).